packages/services/hmi-server/src/main/java/software/uncharted/terarium/hmiserver/resources/simulationservice/SimulationRequestResource.java

```diff
@@ -177,5 +192,42 @@ public JobResponse makeEnsembleCalibrateCiemssJob(
 		return res;
 	}
 
-
+	//Get modelConfigId
+	//Check if it has timeseries in its metadata
+	//If it does for each element convert it to type Intervention and add it to this.interventions
+	//Schema: http://json-schema.org/draft-07/schema#
+	public List<Intervention> getInterventionFromId(String modelConfigId){
+		List<Intervention> interventionList = new ArrayList<>();
+		try {
+			ObjectMapper mapper = new ObjectMapper();
+			ModelConfiguration modelConfig = modelConfigProxy.getModelConfiguration(modelConfigId);
+			JsonNode configuration =  mapper.convertValue(modelConfig.getConfiguration(), JsonNode.class);
+			if (configuration.get("metadata").get("timeseries") != null){
+				JsonNode timeseries = mapper.convertValue(configuration.get("metadata").get("timeseries"), JsonNode.class);
```

Comment:
Because all of these operations are being done on just a `JsonNode` created from another `JsonNode` its a little hard to follow whats happening or what the replacements are doing/for. A comment or two may be good here.. but again not going to block the PR for it

Reply:
Fair ill leave some comments.
This will all be torn out post redesign as the underlying data structure will be fixed up